gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/workunit/packer/KafkaWorkUnitPacker.java

```diff
@@ -362,6 +362,15 @@ public static KafkaWorkUnitPacker getInstance(PackerType packerType, AbstractSou
         throw new IllegalArgumentException("WorkUnit packer type " + packerType + " not found");
     }
   }
+  public double getWorkUnitEstSizes(Map<String, List<WorkUnit>> workUnitsByTopic){
```

Comment:
Is there a reason we can't just use `setWorkUnitEstSizes`? the logic in here is almost identical.

Reply:
The method setWorkUnitEsiSizes is a protected method. So I just create a public method, and this new method just calculating the size instead of setting the property of workUnits

Reply:
You can make the other method public. In general, it is a bad idea to have the same logic duplicated in different locations.

Also, please add javadoc to any public methods.